pkg/scheduler/core/generic_scheduler.go

```diff
@@ -434,21 +441,57 @@ func (g *genericScheduler) prioritizeNodes(
 		return nil, scoreStatus.AsError()
 	}
 
-	if klog.V(10).Enabled() {
-		for plugin, nodeScoreList := range scoresMap {
-			for _, nodeScore := range nodeScoreList {
-				klog.InfoS("Plugin scored node for pod", "pod", klog.KObj(pod), "plugin", plugin, "node", nodeScore.Name, "score", nodeScore.Score)
-			}
-		}
-	}
-
 	// Summarize all scores.
 	result := make(framework.NodeScoreList, 0, len(nodes))
+	totalPluginScores := make(map[string]int64)
 
 	for i := range nodes {
 		result = append(result, framework.NodeScore{Name: nodes[i].Name, Score: 0})
 		for j := range scoresMap {
 			result[i].Score += scoresMap[j][i].Score
+			totalPluginScores[j] += scoresMap[j][i].Score
+		}
+	}
+
+	// Enhanced plugin score logging. The goal of this block is to show the highest scoring plugins on each node,
+	// and the average score for those plugins across all nodes.
+	if klog.V(4).Enabled() {
+		// Build a map of Nodes->PluginScores on that node
+		nodeToPluginScores := make(framework.PluginToNodeScores, len(nodes))
```

Comment:
can you move all of this to its own function?

Reply:
Done